gnosis/cowswap
# 130/cancellation

# Summary

Closes #130 

Soft order cancellation

- Pending orders can now be "soft" cancelled from the "account details" modal
- Added a "cancel" button to pending orders
![screenshot_2021-06-02_15-22-24](https://user-images.githubusercontent.com/43217/120560318-5a848d80-c3b7-11eb-9f65-42f21694ad38.png)
- Cancelled orders are displayed on "account details" modal. They have the same pill color as expired orders, but the icon is the x-circle
- Notification for cancelled orders
![screenshot_2021-06-02_15-27-51](https://user-images.githubusercontent.com/43217/120560316-59ebf700-c3b7-11eb-8d15-46d7d5cff461.png)
- Confirmation for orders to be cancelled
![screenshot_2021-06-02_15-12-57](https://user-images.githubusercontent.com/43217/120560319-5b1d2400-c3b7-11eb-91c7-b6935eaac291.png)
- Refactored operator API HTTP methods
- Added sendOrderCancellation to operator API
- Refactored signature methods
- Added signOrderCancellation to signature methods

# Testing
1. Place an order
2. While it's still pending, click on `Pending` (your account details)
3. On the right of the pending order, click on `cancel`
- [ ]  On the open modal, verify the details match the order you want to cancel
4. Cancel order
5. On your wallet, sign the order cancellation
- [ ] Once it's signed, verify you see a order cancelled toast notification
- [ ] Verify on "account details" modal that the order has a red pill on the left side and red circle with an x on the right side
6. Click on the explorer link for the cancelled order
- [ ] Verify it has the status `cancelled`

## Not included in this PR:
- [x] Improve error handling from API responses (https://github.com/gnosis/cowswap/pull/737)
- [x] Improve styles for cancellable orders on "account details" modal
- [x] Improve styles for cancel order modal
- [x] Improve text for cancel order modal
